File: jscomp/test/condition_compilation_test.ml
let b = 
#if BUFFER_SIZE 
    && 
    BUFFER_SIZE > 20 
      "cool"
#else
      "u"
#end

let buffer_size =
#if true || BUFFER_SIZE 
      1
#else
      32
#end



type open_flag = 
  | O_RDONLY
  | O_WRONLY
  | O_RDWR
  | O_NONBLOCK
  | O_APPEND
  | O_CREAT
  | O_TRUNC
  | O_EXCL
  | O_NOCTTY
  | O_DSYNC
  | O_SYNC
  | O_RSYNC
#if OCAML_VERSION =~ ">=3.13"
  | O_SHARE_DELETE
#end
#if OCAML_VERSION =~ ">=4.01"
  | O_CLOEXEC
#end
#if OCAML_VERSION =~ ">=4.03"
  | O_KEEPEXEC
#end

let vv =
#if OCAML_PATCH = "BS"
      3
#else
      1
#end


let v = ref 1

let a =
#if OCAML_PATCH = "BS"
let () = incr v  in
#end !v

let version_gt_3 =
#if OCAML_VERSION  (* comment *) =~ ">1" 
      true
#else
      false
#end

let version =
#if OCAML_VERSION =~ "~2" 
      2
#elif OCAML_VERSION =~ "~1"
      1
#elif OCAML_VERSION =~ "~0" 
      0
#else
                       -1
#end

let ocaml_veriosn =
#if OCAML_VERSION =~ "~4.02.0" 
      "4.02.3"
#else "unknown"
#end

(**
   #if OCAML_VERSION =~ "4.02.3" #then

   #elif OCAML_VERSION =~ "4.03" #then
   gsho
   #end
*)
(* #if OCAML_VERSION =~ ">4.02" #then *)
(* #else *)
(* #end *)
let suites :  Mt.pair_suites ref  = ref []
let test_id = ref 0
let eq loc x y =
  incr test_id ;
  suites :=
    (loc ^" id " ^ (string_of_int !test_id), (fun _ -> Mt.Eq(x,y))) :: !suites


let () =
  eq __LOC__ vv 3  ;
  eq __LOC__ !v 2


;; Mt.from_pair_suites __MODULE__ !suites

File: jscomp/test/condition_compilation_test.js
'use strict';

var Mt = require("./mt.js");

var v = {
  contents: 1
};

v.contents = v.contents + 1 | 0;

var a = v.contents;

var suites = {
  contents: /* [] */0
};

var test_id = {
  contents: 0
};

function eq(loc, x, y) {
  test_id.contents = test_id.contents + 1 | 0;
  suites.contents = /* :: */{
    _0: [
      loc + (" id " + String(test_id.contents)),
      (function (param) {
          return {
                  tag: /* Eq */0,
                  _0: x,
                  _1: y
                };
        })
    ],
    _1: suites.contents
  };
  
}

eq("File \"condition_compilation_test.ml\", line 98, characters 5-12", 3, 3);

eq("File \"condition_compilation_test.ml\", line 99, characters 5-12", v.contents, 2);

Mt.from_pair_suites("Condition_compilation_test", suites.contents);

var b = "u";

var buffer_size = 1;

var vv = 3;

var version_gt_3 = true;

var version = -1;

var ocaml_veriosn = "unknown";

exports.b = b;
exports.buffer_size = buffer_size;
exports.vv = vv;
exports.v = v;
exports.a = a;
exports.version_gt_3 = version_gt_3;
exports.version = version;
exports.ocaml_veriosn = ocaml_veriosn;
exports.suites = suites;
exports.test_id = test_id;
exports.eq = eq;
/*  Not a pure module */
